### PR TITLE
Move fetch calls out of userDataReducer.js

### LINKED ISSUE #635: Move fetch calls out of userDataReducer.js
https://github.com/uclaacm/TeachLAFrontend/blob/master/src/reducers/userDataReducer.js

Reducers should never contain API calls, which is the case here. Among other problems, error handling in particular is nearly impossible with this setup. This ticket would involve moving the fetch requests to the sources of the action calls. An example is in this file: https://github.com/uclaacm/TeachLAFrontend/blob/master/src/components/common/ProfilePanel.js#L91.
The fetch call for `setDisplayName` should be made here instead of in the reducer. Similar changes should be made to usages of `setPhotoName` and `setMostRecentProgram`.

A perhaps cleaner alternative would be to add middleware along the lines of `redux-thunk` and move the fetch requests to the action creators. This would allow for less code repetition, but increase the number of dependencies.

### PR DESCRIPTION
**Pull request by MDZHX**
Fixes #635.

Fetch calls are moved out of userDataReducer. The rest of the logic (error handling, in particular) remains untouched.

The reason I kept the original logic is that I think it may not be a good idea for me to make too many assumptions about the desired behavior, before any additional discussion/confirmation. In addition, I also saw that there's error-handling logic that depends on the previous implementation in other places. 

That said, we can definitely further discuss any related refactorings in this PR, or we can open a new issue for this.

### CODE REVIEW

File: src/components/Sketches/index.js [hunk 1]
```diff
@@ -51,7 +49,10 @@ class Sketches extends React.Component {
   };
 
   setProgram = (name) => {
-    this.props.setMostRecentProgram(name);
+    const { uid } = this.props;
+
+    // eslint-disable-next-line react/destructuring-assignment
```
Comment (lumisphere902) [inline src/components/Sketches/index.js:54]: I believe this can be removed

File: src/components/common/DropdownButton.js [hunk 2]
```diff
@@ -17,6 +17,7 @@ import {
 
 const DropdownButton = function (props) {
```
Comment (lumisphere902) [inline src/components/common/DropdownButton.js:18]:
```suggestion
function DropdownButton(props) {
```
nit for lint

File: src/components/common/containers/ProfilePanelContainer.js [hunk 3]
```diff
@@ -17,8 +17,34 @@ const mapStateToProps = (state, ownProps) => ({
 const mapDispatchToProps = (dispatch, ownProps) => ({
   onThemeChange: ownProps.onThemeChange,
   collectUserPhoto: () => {},
-  setDisplayName: (name) => dispatch(setDisplayName(name)),
-  setPhotoName: (name) => dispatch(setPhotoName(name)),
+  setDisplayName: (name, uid) => {
+    try {
+      fetch
```
Comment (lumisphere902) [inline src/components/common/containers/ProfilePanelContainer.js:22]: I think a fetch import is missing here

File: src/components/Sketches/containers/ConfirmDeleteModalContainer.js [hunk 4]
```diff
@@ -15,7 +15,18 @@ const mapStateToProps = (state) => {
 
 const mapDispatchToProps = (dispatch) => ({
   deleteProgram: (program, data) => dispatch(deleteProgram(program, data)),
-  setMostRecentProgram: (value) => dispatch(setMostRecentProgram(value)),
+  setMostRecentProgram: (value, uid) => {
+    try {
+      fetch
```
Comment (lumisphere902) [inline src/components/Sketches/containers/ConfirmDeleteModalContainer.js:20]: I think a fetch import is missing here

File: src/components/Sketches/containers/CreateSketchModalContainer.js [hunk 5]
```diff
@@ -1,15 +1,26 @@
...
 const mapStateToProps = (state) => ({
   uid: state.userData.uid,
 });
 
 const mapDispatchToProps = (dispatch) => ({
   addProgram: (program, data) => dispatch(addProgram(program, data)),
-  setMostRecentProgram: (value) => dispatch(setMostRecentProgram(value)),
+  setMostRecentProgram: (value, uid) => {
+    try {
+      fetch
```
Comment (lumisphere902) [inline src/components/Sketches/containers/CreateSketchModalContainer.js:14]: fetch import

**Review by lumisphere902 — CHANGES_REQUESTED** [review]
Hey Zihan, this PR turned out to be pretty big, but you did a great job! There is only one functional issue; the fetch calls shouldn't be working because the fetch library is not currently being imported. I marked some of them, but there are more and I didn't want to make this review super bloated. Other than that, this should be a solid refactor PR!

File: src/components/Sketches/index.js (hunk 1 above)
Comment (MDZHX) [reply src/components/Sketches/index.js:54]: Hi Timothy, thank you for the review! I fixed the other issues just now, but I'm not very sure about what this one is referring to. Do you mean to remove the destructuring of `uid` and use `this.props.setMostRecentProgram(name, this,props.uid)` directly in the next line?

**Comment by lumisphere902** [comment]
I believe the eslint disable is unnecessary.

**Review by lumisphere902 — APPROVED** [review]
Everything looks to be addressed, so feel free to merge at your earliest convenience. The refactor went pretty smoothly, and the overall code consistency also grew with the changes for linting purposes. Good job!

**Pull request merged**